viadot/sources/cloud_for_customers.py

```diff
@@ -0,0 +1,257 @@
+        """Creates url to fetch metadata from.
+
+        Args:
+            url (str): The URL to transform to metadata URL.
+
+        Returns:
+            meta_url (str): The URL to fetch metadata from.
+        """
+        start = url.split(".svc")[0]
+        url_raw = url.split("?")[0]
+        end = url_raw.split("/")[-1]
+        meta_url = start + ".svc/$metadata?entityset=" + end
+        return meta_url
+
+    def _extract_records_from_report_url(self, report_url: str) -> List[Dict[str, Any]]:
+        """Fetches report_url to extract records.
+
+        Args:
+            report_url (str): The url to extract records from.
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from report_url.
+        """
+        records = []
+        while report_url:
+            response = self.get_response(report_url, filter_params=self.filter_params)
+            response_json = response.json()
+            new_records = self.get_entities(response_json, report_url)
+            records.extend(new_records)
+
+            report_url = response_json["d"].get("__next")
+
+        return records
+
+    def _extract_records_from_url(self, url: str) -> List[Dict[str, Any]]:
+        """Fetches URL to extract records.
+
+        Args:
+            url (str): The URL to extract records from.
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from URL.
+        """
+        tmp_full_url = deepcopy(url)
+        tmp_filter_params = deepcopy(self.filter_params)
+        records = []
+        while url:
+            response = self.get_response(tmp_full_url, filter_params=tmp_filter_params)
+            response_json = response.json()
+            if isinstance(response_json["d"], dict):
+                # ODATA v2+ API
+                new_records = response_json["d"].get("results")
+                url = response_json["d"].get("__next", None)
+            else:
+                # ODATA v1
+                new_records = response_json["d"]
+                url = response_json.get("__next", None)
+
+            # prevents concatenation of previous urls with filter_params with the same filter_params
+            tmp_filter_params = None
+            tmp_full_url = url
+
+            if hasattr(new_records, "__iter__"):
+                records.extend(new_records)
+        return records
+
+    def extract_records(self) -> List[Dict[str, Any]]:
+        """Downloads records from `url` or `report_url` if present.
+
+        Returns:
+            records (List[Dict[str, Any]]): The records extracted from URL.
+        """
+        if self.is_report:
+            return self._extract_records_from_report_url(url=self.report_url)
+        else:
+            return self._extract_records_from_url(url=self.full_url)
+
+    def get_entities(
+        self, dirty_json: Dict[str, Any], url: str
+    ) -> List[Dict[str, Any]]:
+        """Extracts entities from request.json().
```

Comment:
What do you mean by entities, some C4C objects? Maybe a link to docs or an example entity could be useful here.

Reply:
I am not sure on this one, since this was already like this, and I found no documentation regarding this one.

Reply:
I will try to guess this following days by printing these objects and trying to understand what entities mean. Also I will read some C4C Documentation, If you have some link to a useful C4C documentation it would be helpful.

Reply:
Done